lib/datapacksexpand.js

```diff
@@ -31,6 +31,10 @@ DataPacksExpand.generateFolderOrFilename = function(filename, extension) {
         .replace(/^[-_\\/]+/, "")
         .replace(/[-_\\/]+$/, "");
 
+    if (sanitizedFilename == '') {
+        sanitizedFilename = 'MissingName';
```

Comment:
We can return something similar like line#22 -- return 'MissingName-' + datapackutils.guid(); And also add a log that why it's missing  the name.